connector/src/main/java/com/google/cloud/hive/bigquery/connector/BigQueryMetaHook.java

```diff
@@ -422,10 +423,28 @@ public void commitInsertTable(Table table, boolean overwrite) throws MetaExcepti
 
   @Override
   public void rollbackInsertTable(Table table, boolean overwrite) throws MetaException {
+    if (overwrite) {
+      try {
+        JobDetails jobDetails =
+            JobDetails.readJobDetailsFile(conf, HiveUtils.getDbTableName(table));
+        if (jobDetails.getTableId() != jobDetails.getFinalTableId()) {
+          // Delete temporary table
+          Injector injector =
```

Comment:
Consider extract a helper method. I see the logic of try-delete-catch is repeated in another place.

Reply:
good catch! added a helper function